client-course-schedulizer/src/components/reuseables/GridItem/GridItemCheckboxGroup/GridItemCheckboxGroup.tsx

```diff
@@ -7,21 +7,46 @@ import {
   FormLabel,
   Grid,
 } from "@material-ui/core";
-import React from "react";
+import React, { ChangeEvent, useEffect, useState } from "react";
 import { useFormContext } from "react-hook-form";
 import { useInput } from "utilities";
 import "./GridItemCheckboxGroup.scss";
 
 interface GridItemCheckboxGroup {
+  initialValue?: string[];
   label: string;
   name?: string;
   options: string[];
-  value?: string[];
 }
 
-export const GridItemCheckboxGroup = ({ label, name, options, value }: GridItemCheckboxGroup) => {
+export const GridItemCheckboxGroup = ({
+  label,
+  name,
+  options,
+  initialValue,
+}: GridItemCheckboxGroup) => {
   const { register, errors } = useFormContext();
   const { name: nameFallback, errorMessage } = useInput(label, errors);
+  const [value, onValueChange] = useState(initialValue);
+
+  useEffect(() => {
+    onValueChange(initialValue);
+  }, [initialValue]);
+
+  const onCheckboxChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const opt = e.target.value;
+    if (value) {
+      if (value?.includes(opt)) {
+        const valueCopy = [...value];
+        valueCopy.splice(valueCopy.indexOf(opt), 1);
+        onValueChange(valueCopy);
+      } else {
+        onValueChange([...value, opt]);
+      }
+    } else {
+      onValueChange([opt]);
```

Comment:
What is this doing? A quick comment would be good.

Reply:
I'll add a comment in #130